src/ExpressReceiver.ts

```diff
@@ -248,42 +233,52 @@ async function verifyRequestSignature(
   hmac.update(`${version}:${ts}:${body}`);
 
   if (!tsscmp(hash, hmac.digest('hex'))) {
-    throw errorWithCode(
-      'Slack request signing verification failed. Signature mismatch.',
-      ErrorCode.ExpressReceiverAuthenticityError,
+    throw new ReceiverAuthenticityError(
+        'Slack request signing verification failed. Signature mismatch.',
     );
   }
 }
 
+/**
+ * This request handler has two responsibilities:
+ * - Verify the request signature
+ * - Parse request.body and assign the successfully parsed object to it.
+ */
+function verifySignatureAndParseBody(
+    signingSecret: string,
+    body: string,
+    headers: Record<string, any>,
+): AnyMiddlewareArgs['body'] {
+  // *** Request verification ***
+  const {
+    'x-slack-signature': signature,
+    'x-slack-request-timestamp': requestTimestamp,
+    'content-type': contentType,
+  } = headers;
+
+  verifyRequestSignature(
+      signingSecret,
+      body,
+      signature,
+      requestTimestamp,
+  );
+
+  return parseRequestBody(body, contentType);
+}
+
 function parseRequestBody(
-  logger: Logger,
-  stringBody: string,
-  contentType: string | undefined): any {
+    stringBody: string,
+    contentType: string | undefined,
+): any {
   if (contentType === 'application/x-www-form-urlencoded') {
     const parsedBody = querystring.parse(stringBody);
+
     if (typeof parsedBody.payload === 'string') {
       return JSON.parse(parsedBody.payload);
     }
-    return parsedBody;
-  }
-
-  if (contentType === 'application/json') {
-    return JSON.parse(stringBody);
-  }
 
-  logger.warn(`Unexpected content-type detected: ${contentType}`);
-  try {
-    // Parse this body anyway
-    return JSON.parse(stringBody);
-  } catch (e) {
-    logger.error(`Failed to parse body as JSON data for content-type: ${contentType}`);
-    throw e;
+    return parsedBody;
   }
 
-}
-
-function receiverAckTimeoutError(message: string): ReceiverAckTimeoutError {
-  const error = new Error(message);
-  (error as ReceiverAckTimeoutError).code = ErrorCode.ReceiverAckTimeoutError;
-  return (error as ReceiverAckTimeoutError);
+  return JSON.parse(stringBody);
```

Comment:
It seemed unnecessary to me, it did a lot of work to add a few extra logs that would only get triggered if Slack itself went haywire or the environment's ingress/proxy/etc was doing likewise. The debugging information didn't seem particularly helpful to me either, it would be strange for one of those ingresses to start converting a contentType on the fly.

Also the global error handler would still log that it couldn't parse. 

CA: 1